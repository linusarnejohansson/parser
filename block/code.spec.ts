import { Error, IO } from "@cogneco/mend"
import { Source } from "../Source"
import { parse } from "."

describe("block.code", () => {
	it("simple", () => {
		const result = parse(new Source(IO.StringReader.create("%%\ncode block\n%%\nFigure Caption."), new Error.ConsoleHandler())) || []
		expect(result.map(node => node.toObject())).toMatchSnapshot()
	})
	it("html", () => {
		const result = parse(new Source(IO.StringReader.create("%% html\n<html>\n<head>\n<title>Title</title>\n</head>\n<body>\n</body>\n</html>\n%%\ncode Caption."), new Error.ConsoleHandler())) || []
		expect(result.map(node => node.toObject())).toMatchSnapshot()
	})
})
