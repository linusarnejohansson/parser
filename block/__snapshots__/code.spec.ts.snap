// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`block.code html 1`] = `
Array [
  Object {
    "class": "Block.Code",
    "content": Array [
      Object {
        "class": "Inline.Text",
        "value": "code Caption.",
      },
    ],
    "language": "html",
    "value": "<html>
<head>
<title>Title</title>
</head>
<body>
</body>
</html>",
  },
]
`;

exports[`block.code simple 1`] = `
Array [
  Object {
    "class": "Block.Code",
    "content": Array [
      Object {
        "class": "Inline.Text",
        "value": "Figure Caption.",
      },
    ],
    "language": "",
    "value": "code block",
  },
]
`;
